h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -62,10 +63,112 @@ else if (!(possibleModel instanceof SharedTreeModel) && !(possibleModel instance
         args.nas = treeProperties._nas;
         args.levels = treeProperties.levels;
         args.predictions = treeProperties._predictions;
-
+        args.tree_decision_path = treeProperties._treeDecisionPath;
+        args.decision_paths = treeProperties._decisionPaths;
         return args;
     }
 
+
+    private static String getLanguageRepresentation(SharedTreeSubgraph sharedTreeSubgraph) {
```

Comment:
I have just noticed that all this code is in Handler class, this is not ideal, we want Handlers to be skinny as possible whith actual logic residing in something thats easier to unit test

Reply:
This is what I did. Not her fault.  It'd a dedicated handler, moving it to other class would result in ..one more class created. Could be cleaner if named accordingly. :wink:

Reply:
as discussed, refactor will be done in a separate PR